chain/rpc/src/client.rs

```diff
@@ -192,30 +192,30 @@ impl RetryPolicy<JsonRpcProviderClientError> for SimpleJsonRpcRetryPolicy {
 
         // Retry if a global minimum of number of retries was given and wasn't yet attained
         if self.min_retries.is_some_and(|min| num_retries <= min) {
-            debug!("retrying because {num_retries} is not yet minimum number of retries");
+            debug!(num_retries, min_retries = ?self.min_retries,  "retrying because minimum number of retries not yet reached");
             return RetryAfter(backoff);
         }
 
         match err {
             // Retryable JSON RPC errors are retries with backoff
             JsonRpcProviderClientError::JsonRpcError(e) if self.is_retryable_json_rpc_error(e) => {
-                debug!("encountered retryable JSON RPC error code: {e}");
+                debug!(error = ?e, "encountered retryable JSON RPC error code");
                 RetryAfter(backoff)
             }
 
             // Retryable HTTP errors are retries with backoff
             JsonRpcProviderClientError::BackendError(HttpRequestError::HttpError(e))
                 if self.is_retryable_http_error(e) =>
             {
-                debug!("encountered retryable HTTP error code: {e}");
+                debug!(error = ?e, "encountered retryable HTTP error code");
                 RetryAfter(backoff)
             }
 
             // Transport error and timeouts are retried at a constant rate if specified
             JsonRpcProviderClientError::BackendError(e @ HttpRequestError::Timeout)
             | JsonRpcProviderClientError::BackendError(e @ HttpRequestError::TransportError(_))
             | JsonRpcProviderClientError::BackendError(e @ HttpRequestError::UnknownError(_)) => {
-                debug!("encountered retryable transport error: {e}");
+                debug!(error = ?e, "encountered retryable transport error");
```

Comment:
```suggestion
                debug!(error = %e, "encountered retryable transport error");
```

Reply:
We don't want the `Debug` version, but the `Display` version of the error in the log output.